addOns/llm/src/main/java/org/zaproxy/addon/llm/utils/Requestor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.llm.utils;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.addon.llm.communication.HttpRequest;
+import org.zaproxy.addon.llm.communication.HttpRequestList;
+
+public class Requestor {
+    private static final Logger LOGGER = LogManager.getLogger(Requestor.class);
+
+    private int initiator;
+    private HistoryPersister listener;
+    private HttpSender sender;
+
+    public Requestor(int initiator, HistoryPersister listener) {
+        this.initiator = initiator;
+        this.listener = listener;
+        this.sender = new HttpSender(initiator);
+    }
+
+    public String getResponseBody(HttpRequest httpRequest) throws IOException {
+        HttpMessage httpMessage = new HttpMessage();
+        httpMessage.setRequestHeader(
+                httpRequest.getMethod() + " " + httpRequest.geturl() + " HTTP/1.1");
```

Comment:
I believe the core has constants for HTTP version strings. Check HttpHeader or HttpRequestHeader

Reply:
Could be used down on L61 too